Library/Homebrew/formulary.rb

```diff
@@ -160,17 +160,17 @@ def initialize tapped_name
       @tapped_name = tapped_name
       user, repo, name = tapped_name.split("/", 3).map(&:downcase)
       tap = Pathname.new("#{HOMEBREW_LIBRARY}/Taps/#{user}/homebrew-#{repo}")
-      path = tap.join("#{name}.rb")
 
       if tap.directory?
         tap.find_formula do |file|
           if file.basename(".rb").to_s == name
-            path = file
+            super name, file
+            return
           end
         end
       end
 
-      super name, path
+      raise TapFormulaUnavailableError, tapped_name
     end
```

Comment:
Make sure either returning the founded tapped formula or raising error. There will be no state that to return the loader but the path doesn't exist. Therefore, we guarantee  that path always existing to downstream.